ow_dynamic_terrain/src/DynamicTerrainBase.cpp

```diff
@@ -0,0 +1,68 @@
+#include "DynamicTerrainBase.h"
+#include "memory_ext.h"
+
+using namespace std;
+using namespace ow_dynamic_terrain;
+
+void DynamicTerrainBase::Initialize(const std::string& topic_extension)
+{
+  if (!ros::isInitialized())
+  {
+    gzerr << m_plugin_name + ": ROS not initilized! The plugin won't load" << endl;
+    return;
+  }
+
+  m_node_handle = make_unique<ros::NodeHandle>(m_plugin_name);
+  m_node_handle->setCallbackQueue(&m_callback_queue);
+
+  auto on_modify_terrain_circle = [this](const modify_terrain_circle::ConstPtr& msg) {
+    this->onModifyTerrainCircleMsg(msg);
+  };
+  subscribe<modify_terrain_circle>("modify_terrain_circle", on_modify_terrain_circle);
+  subscribe<modify_terrain_circle>("modify_terrain_circle/" + topic_extension, on_modify_terrain_circle);
+
+  auto on_modify_terrain_ellipse = [this](const modify_terrain_ellipse::ConstPtr& msg) {
+    this->onModifyTerrainEllipseMsg(msg);
+  };
+  subscribe<modify_terrain_ellipse>("modify_terrain_ellipse", on_modify_terrain_ellipse);
+  subscribe<modify_terrain_ellipse>("modify_terrain_ellipse/" + topic_extension, on_modify_terrain_ellipse);
+
+  auto on_modify_terrain_patch = [this](const modify_terrain_patch::ConstPtr& msg) {
+    this->onModifyTerrainPatchMsg(msg);
+  };
+  subscribe<modify_terrain_patch>("modify_terrain_patch", on_modify_terrain_patch);
+  subscribe<modify_terrain_patch>("modify_terrain_patch/" + topic_extension, on_modify_terrain_patch);
+
+  m_on_update_connection = gazebo::event::Events::ConnectPostRender([this]() {
+    if (m_node_handle->ok())
+      m_callback_queue.callAvailable();
+  });
+
+  gzlog << m_plugin_name << ": successfully loaded!" << endl;
+}
+
+template <typename T>
+void DynamicTerrainBase::subscribe(const std::string& topic,
+                                   const boost::function<void(const boost::shared_ptr<T const>&)>& callback)
+{
+  auto topic_fqn = "/" + m_package_name + "/" + topic;
```

Comment:
Please use simple type names instead of auto.

Reply:
addressed